erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -690,6 +691,21 @@ def update_stock_ledger(self):
 
 		self.make_sl_entries(sl_entries, self.amended_from and 'Yes' or 'No')
 
+	def validate_mandatory_batch_number(self):
+		"""
+			validate the stock items batch number is present or not
+		"""
+		items_without_batch_no = []
+
+		for idx, item in enumerate(self.get("items"), start=1):
```

Comment:
with enumeration, the index 0 gives the first element, ideally which is correct, but we want to display the error msg starting from count= 1, not 0. Thus have added start=1, which starts count with the value 1 on the first loop iteration.
Refer below image:-
![image](https://user-images.githubusercontent.com/80836439/124603717-da56b980-de87-11eb-99a5-89f93dc2f823.png)